create-plugin-update/action.yml

```diff
@@ -0,0 +1,85 @@
+name: "Grafana Build Plugin"
+description: "Builds a Grafana plugin"
+
+outputs:
+  archive:
+    description: "The path to the plugin archive (zip)."
+    value: ${{ steps.package-plugin.outputs.archive }}
+  archive-sha1sum:
+    description: "The path to the plugin archive sha1sum."
+    value: ${{ steps.package-plugin.outputs.archive-sha1sum }}
+
+inputs:
+  token:
+    description: "Token for the repository. Can be passed in using `{{ secrets.GITHUB_TOKEN }}`."
+    required: false
+    default: "${{ github.token }}"
+  base:
+    description: "Sets the pull request base branch"
+    required: false
+    default: "main"
+  node-version:
+    description: "Version of node"
+    required: false
+    default: "20"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: "${{ inputs.node-version }}"
+
+    - name: Get latest version of create-plugin package
+      id: get-latest-version
+      run: echo "LATEST_VERSION=$(npx -y @grafana/create-plugin@latest version)" >> "$GITHUB_ENV"
+      shell: bash
+
+    - name: Checkout Repository
+      uses: actions/checkout@v4
+      env:
+        GITHUB_TOKEN: ${{ inputs.token }}
+
+    - name: Get version from .config/.cprc.json
+      id: get-config-version
+      run: |
+        CONFIG_VERSION=$(jq -r '.version' .config/.cprc.json)
+        echo "CONFIG_VERSION=${CONFIG_VERSION}" >> "$GITHUB_ENV"
+      shell: bash
+
+    - name: Compare versions
+      id: compare-versions
+      run: |
+        if [ "$LATEST_VERSION" != "$CONFIG_VERSION" ]; then
+          echo "update_needed=true" >> $GITHUB_OUTPUT
+        else
+          echo "update_needed=false" >> $GITHUB_OUTPUT
+        fi
+      shell: bash
+
+    - name: Update the configs
+      if: steps.compare-versions.outputs.update_needed == 'true'
+      run: |
+        ${{ github.action_path }}/pm.sh update
+        ${{ github.action_path }}/pm.sh install
+      shell: bash
+
+    - name: Create PR
+      uses: peter-evans/create-pull-request@v7
+      if: steps.compare-versions.outputs.update_needed == 'true'
+      with:
+        token: ${{ inputs.token }}
+        base: ${{ inputs.base }}
+        branch: update-grafana-create-plugin
+        delete-branch: true
+        commit-message: "chore: update configuration to create-plugin ${{ env.LATEST_VERSION }}"
+        title: "Chore: Bump @grafana/create-plugin configuration to ${{ env.LATEST_VERSION }}"
+        author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
```

Comment:
is this author accurate? (Shouldn't that be inferred from the token?)

Reply:
I don't know if it's accurate but we cannot push unless the use name and email are set in the git config. Using the github bot seemed like a reasonable default choice. We could offer it as an option if devs would like to customise it?